frontend/packages/ux-editor/src/components/config/editModal/EditOptions/EditCodeList/EditCodeList.test.tsx

```diff
@@ -182,45 +130,67 @@ describe('EditCodeList', () => {
     const file = new File([optionListIdsMock[0]], invalidFileName, {
       type: 'text/json',
     });
-    await render({
-      queries: {
-        getOptionListIds: jest
-          .fn()
-          .mockImplementation(() => Promise.resolve<string[]>(optionListIdsMock)),
-      },
-    });
 
-    const btn = screen.getByRole('button', {
-      name: textMock('ux_editor.modal_properties_code_list_upload'),
-    });
-    await user.click(btn);
-
-    const fileInput = screen.getByLabelText(
-      textMock('ux_editor.modal_properties_code_list_upload'),
-    );
-    await user.upload(fileInput, file);
+    renderEditCodeList();
+    await userFindUploadButtonAndClick(user);
+    await userFindFileAndUpload(user, file);
 
     expect(await screen.findByRole('alert')).toHaveTextContent(
       textMock('ux_editor.model_properties_code_list_filename_error'),
     );
   });
+
+  it('should render OptionListEditor when featureFlag is active', async () => {
+    addFeatureFlagToLocalStorage('codeListEditor');
+    const queries = {
+      getOptionListIds: jest
+        .fn()
+        .mockImplementation(() => Promise.resolve<string[]>(optionListIdsMock)),
+      getOptionLists: jest
+        .fn()
+        .mockImplementation(() =>
+          Promise.resolve<Map<string, Option[]>>(
+            new Map<string, Option[]>([['text', [{ value: 'test', label: 'label text' }]]]),
+          ),
+        ),
+    };
+    renderEditCodeList({ queries: queries });
+
+    await waitForElementToBeRemoved(() =>
+      screen.queryByText(textMock('ux_editor.modal_properties_code_list_spinner_title')),
+    );
+    expect(
+      screen.getByRole('button', {
+        name: textMock('ux_editor.modal_properties_code_list_open_editor'),
+      }),
+    ).toBeInTheDocument();
+  });
 });
 
-const render = async ({
-  handleComponentChange = jest.fn(),
-  queries = {},
-  componentProps = {},
-} = {}) => {
-  renderWithProviders(
+const userFindUploadButtonAndClick = async (user: UserEvent) => {
+  const btn = screen.getByRole('button', {
+    name: textMock('ux_editor.modal_properties_code_list_upload'),
+  });
+  await user.click(btn);
+};
+
+const userFindFileAndUpload = async (user: UserEvent, file: File) => {
+  const fileInput = screen.getByLabelText(textMock('ux_editor.modal_properties_code_list_upload'));
+
+  await user.upload(fileInput, file);
+};
+
+const renderEditCodeList = ({ queries = queriesMock, componentProps = {} } = {}) => {
+  return renderWithProviders(
```

Comment:
It does not matter but I think a more usual pattern in our codebase is to create the queryClientMock here and not in the afterEach